### PR TITLE
Fix numpy 117 rc and dask in thresholding filters

### PR DESCRIPTION
**Pull request by hmaarrfk**
## Description

Fixes tests for numpy 1.17 and dask.

I get a really strange warning from dask and true divide, but I'm note sure where it is from.

As far as I can tell, the warning is reproduced by

```python
import numpy as np
import skimage
import dask
from skimage.exposure import histogram
from dask import array as da
from skimage import data
print(f"numpy version {np.__version__}")
print(f"scikit-image version {skimage.__version__}")
print(f"dask version {dask.__version__}")
image = da.from_array(data.camera(), chunks=(256, 256))
nbins = 256

hist, bin_centers = histogram(image.ravel(), nbins, source_range='image')
nbins = len(hist)

# Find peak, lowest and highest gray levels.
arg_peak_height = np.argmax(hist)
peak_height = hist[arg_peak_height]
arg_low_level, arg_high_level = np.where(hist>0)[0][[0, -1]]



# Find peak, lowest and highest gray levels.
arg_peak_height = np.argmax(hist)
peak_height = hist[arg_peak_height]
arg_low_level, arg_high_level = np.where(hist>0)[0][[0, -1]]


# Flip is True if left tail is shorter.
flip = (arg_peak_height - arg_low_level) < (arg_high_level - arg_peak_height)
if flip:
    hist = hist[::-1]
    arg_low_level = nbins - arg_high_level - 1
    arg_peak_height = nbins - arg_peak_height - 1


# If flip == True, arg_high_level becomes incorrect
# but we don't need it anymore.
del(arg_high_level)



# Set up the coordinate system.
width = arg_peak_height - arg_low_level
x1 = np.arange(width)
y1 = hist[x1 + arg_low_level]


norm = np.sqrt(peak_height**2 + width**2)

peak_height / norm

"""
/home/mark/miniconda3/envs/skdev/lib/python3.7/site-packages/dask/array/core.py:3808: RuntimeWarning: invalid value encountered in true_divide
  result = function(*args, **kwargs)
dask.array<truediv, shape=(), dtype=float64, chunksize=()>
"""

norm

norm.compute()
"""6216.711992685522"""

peak_height.compute()
"""6212"""
```

Should open an issue upstream

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by hmaarrfk** [comment]
This is a bug in dask

```
from dask import array as da
import numpy as np

a = da.ones((1, ), chunks=(1,))[0]
1 / np.sqrt(a)

```

gives the same warning. There is no division by zero here

**Comment by hmaarrfk** [comment]
I guess i also sprinkled some optimization in the code too.

**Comment by jni** [comment]
@hmaarrfk super awesome!

@jakirkham if you have a chance to review this now, I'd appreciate it! Otherwise will get to it shortly, having some chats, thank you @hmaarrfk!

File: skimage/filters/thresholding.py [hunk 1]
```diff
@@ -434,11 +434,12 @@ def threshold_isodata(image, nbins=256, return_all=False):
     # last bin of csumh, which is zero by construction.
     # So no worries about division by zero in the following lines, except
     # for the last bin, but we can ignore that because no valid threshold
-    # can be in the top bin. So we just patch up csumh[-1] to not cause 0/0
-    # errors.
-    csumh[-1] = 1
```
Comment (hmaarrfk) [inline skimage/filters/thresholding.py]: The issue stemmed from this line of code. You can't assign something to dask  I just optimized that line away all together.

**Comment by jni** [comment]
@hmaarrfk thank you! So awesome!

> This is a bug in dask

have you reported it upstream? Seems important!

**Comment by hmaarrfk** [comment]
I did. It is referenced above. It is so fundamental I can't believe it exists....

**Pull request merged**